rfd/0128-aws-integration-tests.md

```diff
@@ -0,0 +1,191 @@
+to generate a short-lived token that is used to authenticate with the Kubernetes
+API.
+
+The integration tests for Kubernetes access will test the following:
+
+- The discovery service can discover Kubernetes clusters running on AWS and
+  create the `kube_cluster` object in the Teleport database. If the object
+  already exists, it will be updated with the new labels.
+- The Kubernetes service receives the `kube_cluster` object and can generate
+  a short-lived token to authenticate with the Kubernetes API.
+- The Kubernetes service can forward the requests to the Kubernetes API.
+- The Kubernetes service automatically refreshes the token when it's about to
+  expire.
+
+#### Requirements
+
+This section describes the requirements for the integration tests for Kubernetes
+access.
+
+- One or more EKS control plane clusters running on AWS. We don't need that the
+  clusters are running any workloads, we only need that their control plane is
+  running and accessible.
+- Discovery service configured to discover the EKS clusters and with List and Describe
+  permissions to the EKS API. [Permissions](https://goteleport.com/docs/kubernetes-access/discovery/aws/#step-13-set-up-aws-iam-credentials)
+- Kubernetes service IAM role configured with the minimum required permissions
+  to generate the short-lived token and forward the requests to the Kubernetes
+  API. [IAM Mapping](https://goteleport.com/docs/kubernetes-access/discovery/aws/#iam-mapping)
+
+Spin up the EKS clusters takes a long time and it's not feasible to do it for
+each test. To speed up the process, we should use an existing EKS cluster that
+is already running, accessible and configured to allow Teleport Kubernetes service
+to access it. This will allow us to run the integration tests without having to
+wait for the EKS cluster to be created. Since we can interact with the EKS API
+and we do not need to run any workloads on the cluster, the existing EKS cluster
+does not need to have dedicated nodes to run workloads. A single control plane
+deployed on a single availability zone is enough.
+
+## Security
+
+Several security considerations must be to be taken into account
+when implementing the integration tests for AWS.
+
+### GitHub OIDC Provider
+
+GitHub OIDC provider will be used to authenticate with AWS API. This will allow
+us to use GitHub actions to run the integration tests without having to manage
+AWS credentials. GitHub actions will be allowed to assume a set of roles that
+will simulate the minimum required permissions that each Teleport service
+requires to perform the required actions. An important consideration is that
+all other OIDC tokens from GitHub that don't belong to the Teleport Enterprise
+Account actions or originated from other repositories besides `gravitational/teleport`
+must be rejected. This includes tokens from other GitHub
+organizations and tokens from the Teleport Enterprise Account that don't belong
+to the GitHub actions pipeline.
```

Comment:
Does it mean that single individual user won't be able to run these test on local machine  ? I think that we should consider a way to  provide a fallback mechanism and allowing a user outside Teleport Organisation to run this test.

Reply:
This only specifies the scope for the Github action flows. 

Individuals with access to Teleport AWS accounts can always create ec2 instances with the associated IAM policies for testing and debugging purposes.